### ISSUE TITLE
Question about the usage of optical flow, scene flow, and point tracking

### ISSUE THREAD

**Issue by ShenhanQian** [eugenelyj/blinkvision_benchmark#7]
Dear Authors,

Thank you for open-sourcing this excellent dataset. I have several question regarding the usage of pixel-wise correspondences:

### Optical Flow

I noticed that there are several folders related to optical flow, including:
- `forward_folder`
- `forward_folder_custom_stride`
- `backward_folder`
- `backward_folder_custom_stride`

My initial assumption was that  `forward_flow/000000.npz` would be identical to `forward_flow_custom_stride/000000_000001.npz`. However, I found some differences:
- `forward_flow/000000.npz` contains an  np.flow32 array in shape (540, 960, 2)
-  `forward_flow_custom_stride/000000_000001.npz` contains an np.float64 array in shape (540, 960, 3).

Additionally, visualizing the first two channels of these arrays gives noticeably different results (see attached figures below):
<img width="668" height="580" alt="Image" src="https://github.com/user-attachments/assets/7e509c33-0010-4bf9-825d-e0abfe592422" />
<img width="668" height="580" alt="Image" src="https://github.com/user-attachments/assets/489dea8f-f147-4c27-838c-97afdca96b83" />

It’s possible that I have misunderstood the intended data definition. Could you kindly clarify the differences between these optical flow folders?

### Scene Flow
Is scene flow already included in the released data?

### Point Tracking
The point tracking annotations are stored in `particle/`, each contain an np.float32 array of shape (540, 960, 4). I assume that the first three channels are xyz-coordinates. But is the last channel for confidence/validity or occlusion flag? Meanwhile, are the particles defined in the world space or the camera space?

Thank you very much for your time and for providing this resource.

**Comment by eugenelyj** [comment]
@ShenhanQian Sorry for the late reply.
1. Could you tell me which scene you used? I have sampled some data from forward_flow and forward_flow_custom_stride and they are almost the same. The difference should be not big and it comes from the difference in computation which you can refer to the  [simulation code](https://github.com/zju3dv/blink_sim/blob/v2/src/blender/blender_script.py#L275). The forward_flow is computed through in-built blender API and the forward_flow_custom_stride is computed in the same way as the particle and then converted to optical flow.
2. The scene flow is actually the forward_flow_custom_stride that has shape [h, w, 3].
3. The four channels are xyz, and status. xyz is in the camera space. The status includes the following values. 
particle_VISIBLE = 1
particle_OCCLUDED = 0
particle_OUTBOUND = -1
particle_NO_CAST = -2
particle_BEHIND_CAMERA = -3
You can refer to the [simulation code](https://github.com/zju3dv/blink_sim/blob/v2/src/blender/particle.py#L51) and the [visualization code](https://github.com/eugenelyj/blinkvision_benchmark/blob/main/blinkvision_v1/pytorch_dataloader/train_dataset.py#L350)

**Comment by ShenhanQian** [comment]
@eugenelyj Thanks for your reply.

1. Below is the script I use to visualize the optical flow:
    ```python
    import numpy as np
    
    scene = 'BlinkVision/train/indoor_train/AI33_002_280_people'
    
    f = np.load(f'{scene}/forward_flow/000000.npz')['forward_flow']  # H, W, 2
    fc = np.load(f'{scene}/forward_flow_custom_stride/000000_000001.npz')['forward_flow']  # H, W, 3
    ffc = np.concatenate([f, fc[:, :, :2]], axis=1)   # stack horizontally to ease comparison
    
    # visualize with torchshow
    import torchshow as ts  # pip install torchshow
    ts.show(ffc)   # torchshow automatically visualize in optical flow mode for 2-channel images
    # ts.save(ffc, 'vis_ts.png')  # use this in case ts.show does not work on headless servers
    
    # visualize with matplotlib
    import matplotlib.pyplot as plt
    ffc_pad = np.pad(ffc, [0, 1])  # maplotlib requires 3-channel images
    ffc_pad = ffc_pad / np.abs(ffc_pad).max() / 2 + 0.5
    plt.imshow(ffc_pad)
    plt.show()
    # plt.savefig('vis_mp.png')  # use this in case plt.show does not work on headless servers
    ```

    In this example, there exist some inconsistency in the the foreground

    <img width="1495" height="535" alt="Image" src="https://github.com/user-attachments/assets/fb07fd35-0b80-47a0-8b09-e0ad361155e7" />

2. By "scene flow", I mean the 3D translation vector of a 3D point corresponding to each pixel. But `ffc` read from `forward_flow_custom_stride` in the above script always contains 1 in the last channel. This conflicts with my understanding of scene flow. Could you please elaborate a bit on this?

3. Thanks for the detailed list!

**Comment by eugenelyj** [comment]
@ShenhanQian Hi, thanks for the information.
1. I checked it and it showed as you pointed out. The difference is on the human body. I have no idea now and need some time to check what happened. Previously, I have simply checked the particle data through visualization and the custom flow data is converted from particle so I think it have no problem. I didn't check the forward flow data from blender API before. I will double check the data through warping and tell you if I have conclusions.
2. You are right. I missed the scene flow data when uploading. I am unzip these data and will upload it in about two days.

**Comment by ShenhanQian** [comment]
Thank you very much for the clarification! Looking forward to the scene flow data being uploaded soon 💪

**Comment by ShenhanQian** [comment]
Hey @eugenelyj, may I ask if the scene flow is ready to download somewhere?

**Comment by eugenelyj** [comment]
@ShenhanQian Sorry for the late reply. Unfortunately, the scene flow training data is not available now. I unzip the scene flow folder last weak but after checking, I found they are elder version that share some assets with the testing data so I can not upload them. For the correct version, I didn't find them right now. Despite this, you can indirectly obtain scene flow training data in the following two ways. The first way comes from the particle data and the second way comes from the flow and depth data. Particle data cover all information of scene flow but these data is sampled more sparse due to the visible overlap strategy that is introduced in the paper. In contrast, the second way give us more densely sampled data but the z-axis of scene flow is missing for those occluded pixels.

**Comment by ShenhanQian** [comment]
Hi @eugenelyj Thanks for the instant reply! I might go with the particle data to see if it works for my use case. Cheers!
